src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -2792,7 +2795,8 @@ public <V> V runWithCompactionsDisabled(Callable<V> callable,
     {
         // synchronize so that concurrent invocations don't re-enable compactions partway through unexpectedly,
         // and so we only run one major compaction at a time
-        synchronized (this)
+        rwcdLock.lock();
```

Comment:
@blambov let me pick your brain here. We're getting deadlocks here when syncing on CF and later on the strategy when pausing or resuming compactions. Then a UCS task locks on the strategy first an later on CF. Trying to reorder locking produces all sorts of new deadlocks elsewhere in the code.

Given `runWithCompactionsDisabled` syncs on CF to avoid concurrent calls iiuc, do you see any problems with this approach?
- Use an aux lock to avoid concurrent calls
- Reorder the sync against CF to come later avoiding nested and badly ordered syncs

I am just wary the original sync against CF was guarding against sthg else I might be missing.